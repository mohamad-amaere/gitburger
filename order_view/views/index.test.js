const ejs = require("ejs");
const fs = require("fs");
const path = require("path");

describe("index.ejs", () => {
	it("should match the snapshot", async () => {
		const filePath = path.join(__dirname, "index.ejs");
		const template = fs.readFileSync(filePath, "utf-8");

		const data = {
			menu: [
				{ id: 1, burger_name: "Cheeseburger", price: "$5.99" },
				{ id: 2, burger_name: "Veggie Burger", price: "$6.99" }
			],
			cart: []
		};

		// Mock the include function to avoid errors
		const options = {
			views: [path.join(__dirname, "partials")],
			filename: path.join(__dirname, "index.ejs"),
			context: {
				include: (filePath) => {
					const partialPath = path.join(
						__dirname,
						"partials",
						`${filePath}.ejs`
					);
					if (fs.existsSync(partialPath)) {
						return fs.readFileSync(partialPath, "utf-8");
					}
					return "";
				}
			}
		};

		// Render the template
		const html = await ejs.renderFile(
			path.join(__dirname, "thankyou.ejs"),
			data,
			options
		);
		expect(html).toMatchSnapshot();
	});
});
