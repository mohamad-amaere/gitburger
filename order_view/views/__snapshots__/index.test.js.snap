// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.ejs should match the snapshot 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@20..48,100..700,0..1,-50..200&icon_names=shopping_cart" />

<style>
.material-symbols-outlined {
  font-variation-settings:
  'FILL' 0,
  'wght' 400,
  'GRAD' 0,
  'opsz' 24
}
</style>

</head>
<body>
  <style>
  nav {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 10px;
    background-color: lightblue;
  }

  ul {
    display: flex;
    list-style-type: none;
  }

  li {
    margin-right: 10px;
    background-color: #f0f0f0;
    padding: 1rem .5rem;
    border-radius: 10%;
  }

  li:hover {
    background-color: #e0e0e0;
  }

  a {
    text-decoration: none;
    color: black;
  }

  div {
    display: flex;
    align-items: center;
  }

  .material-symbols-outlined {
    font-variation-settings:
      'FILL' 0,
      'wght' 400,
      'GRAD' 0,
      'opsz' 24
  }

  .amountInCart {
    margin-left: 5px;
  }

  .cart-dialog {
    display: none;
    position: absolute;
    top: 50px;
    right: 10px;
    width: 300px;
    background-color: white;
    border: 1px solid #ccc;
    box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
    padding: 10px;
    z-index: 1000;
  }

  .cart-dialog.active {
    display: block;
  }
</style>

<nav>
  <ul>
    <li><a href="/">Home</a></li>
  </ul>

  <div>
    <span class="material-symbols-outlined" id="cart-button">
      shopping_cart
    </span>
    <span class="amountInCart">0</span>
  </div>
</nav>

<div class="cart-dialog" id="cart-dialog">
  <h3>Your Cart</h3>
  <ul>
    
  </ul>
  <p>total: 0 Kr
  <form method="post" action="/order">
    <button type="submit">Place order</button>
  </form>
</div>

<script>
  document.getElementById('cart-button').addEventListener('click', function() {
    var cartDialog = document.getElementById('cart-dialog');
    cartDialog.classList.toggle('active');
  });

  document.addEventListener('click', function(event) {
    var cartDialog = document.getElementById('cart-dialog');
    if (!cartDialog.contains(event.target) && event.target.id !== 'cart-button') {
      cartDialog.classList.remove('active');
    }
  });
</script>

  <h1>Thank you for your order</h1>
</body>
</html>"
`;
